lib/ansible/modules/tacp/tacp_info.py

```diff
@@ -0,0 +1,198 @@
+    api_client = tacp.ApiClient(configuration)
+
+    resource_dict = {"application": tacp_utils.ApplicationResource,
+                     "application_group": tacp_utils.ApplicationGroupResource,
+                     "category": tacp_utils.CategoryResource,
+                     "datacenter": tacp_utils.DatacenterResource,
+                     "firewall_profile": tacp_utils.FirewallProfileResource,
+                     "instance": tacp_utils.ApplicationResource,
+                     "marketplace_template": tacp_utils.MarketplaceTemplateResource,
+                     "migration_zone": tacp_utils.MigrationZoneResource,
+                     "site": tacp_utils.SiteResource,
+                     "storage_pool": tacp_utils.StoragePoolResource,
+                     "tag": tacp_utils.TagResource,
+                     "template": tacp_utils.TemplateResource,
+                     "user": tacp_utils.UserResource,
+                     "vlan": tacp_utils.VlanResource,
+                     "vnet": tacp_utils.VnetResource
+                     }
+
+    resource = resource_dict[module.params['resource']](api_client)
+
+    items = [item.to_dict() for item in resource.filter()]
+
+    if module.params['resource'] == 'migration_zone':
+        application_resource = tacp_utils.ApplicationResource(
+            api_client)
+        category_resource = tacp_utils.CategoryResource(
+            api_client)
+        datacenter_resource = tacp_utils.DatacenterResource(
+            api_client)
+        for item in items:
+            if 'applications' in item:
+                for application in item['applications']:
+                    application['name'] = application_resource.get_by_uuid(
+                        application['uuid']).to_dict()['name']
+
+            if 'allocations' in item:
+                if 'categories' in item['allocations']:
+                    for category in item['allocations']['categories']:
+                        category['name'] = category_resource.get_by_uuid(
+                            category['category_uuid']).to_dict()['name']
+
+                if 'datacenters' in item['allocations']:
+                    for datacenter in item['allocations']['datacenters']:
+                        datacenter['name'] = datacenter_resource.get_by_uuid(
+                            datacenter['datacenter_uuid']).to_dict()['name']
+
+    elif module.params['resource'] == 'datacenter':
+        vnet_resource = tacp_utils.VnetResource(api_client)
+        vlan_resource = tacp_utils.VlanResource(api_client)
+        tag_resource = tacp_utils.TagResource(api_client)
+        for item in items:
+            if 'networks' in item:
+                for network in item['networks']:
+                    try:
+                        network['name'] = vnet_resource.get_by_uuid(
+                            network['uuid']).to_dict()['name']
+                        network['network_type'] = 'vnet'
+                    except Exception:
```

Comment:
What kind of error you are trying to catch here?

Reply:
Found a better solution, not doing this anymore :)